File: 1.两数之和.cpp
/*
 * @lc app=leetcode.cn id=1 lang=cpp
 *
 * [1] 两数之和
 */

// @lc code=start
#include<unordered_map>
#include<algorithm>
#include<iostream>
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int,int> dict;
        for(int i=0;i<nums.size();++i){
            if(dict.count(nums[i])){
                return {dict[nums[i]],i};
            }
            else{
                dict[target-nums[i]]=i;
            }
        }
        return {};
    }
};
// @lc code=end

File: 1.两数之和.py
#
# @lc app=leetcode.cn id=1 lang=python3
#
# [1] 两数之和
#

# @lc code=start
from typing import List
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dp = {}
        for idx, elem in enumerate(nums):
            if elem in dp: return [dp[elem], idx]
            dp[target - elem] = idx
# @lc code=end

print(Solution().twoSum(nums = [2,7,11,15], target = 9))
print(Solution().twoSum(nums = [3,2,4], target = 6))
print(Solution().twoSum(nums = [3,3], target = 6))